distributed/protocol/cupy.py

```diff
@@ -80,3 +80,20 @@ def dask_deserialize_cupy_ndarray(header, frames):
     frames = [dask_deserialize_cuda_buffer(header, frames)]
     arr = cuda_deserialize_cupy_ndarray(header, frames)
     return arr
+
+
+try:
+    from cupy.cusparse import MatDescriptor
+    from cupyx.scipy.sparse import spmatrix
+
+    cupy_sparse_types = [MatDescriptor, spmatrix]
+except ImportError:
```

Comment:
What is the behavior here when the import fails.  Will there be no serialization defined for cupy sparse arrays ?

Reply:
Presumably in this case they don't exist.

This seems fine to me.  I might write it slightly differently, but this is just a subjective/style difference and not a difference worth spending much time on.

```python
try:
    import ...
except ImportError:
    pass
else:
    for typ in [MatDescriptor, spmatrix]:
        ...
```

Reply:
If the `import` fails, we don't register these. 🙂

Yeah `else` as part of `try...except...finally...else` and `for...else` has bitten me a few times in unexpected ways before. Would prefer to avoid it as it is not really any simpler and the failure modes are annoying to debug.

Reply:
Ah ok. if the import fails then nothing is registered.  Sorry, that took me a minute.